plugins/cache-ehcache/src/main/java/org/opensearch/cache/store/disk/EhcacheDiskCache.java

```diff
@@ -176,6 +186,8 @@ public Duration getExpiryForUpdate(K key, Supplier<? extends V> oldValue, V newV
                             (Integer) EhcacheDiskCacheSettings.getSettingListForCacheType(cacheType).get(DISK_SEGMENT_KEY).get(settings)
                         )
                     )
+                    .withKeySerializer(new KeySerializerWrapper<K>(keySerializer))
+                    .withValueSerializer(new ByteArrayWrapperSerializer())
```

Comment:
Could we not use ```new ByteArrayWrapperSerializer(valueSerializer)``` now?
If not, maybe just add a reason somewhere of doing this way.

Reply:
The value serializer has to be outside of ehcache itself, because ehcache enforces that the class be exactly the same before and after serialization. This isn't feasible for an interface like ByteReference. So ByteArrayWrapperSerializer just consumes ByteArrayWrapper directly and implements the necessary ehcache interfaces. I'll add a comment explaining this